packages/context-menu/src/vaadin-menu-overlay-mixin.js

```diff
@@ -116,27 +121,89 @@ export const MenuOverlayMixin = (superClass) =>
       if (this.positionTarget && this.parentOverlay) {
         // This overlay is positioned by a parent menu item,
         // adjust the position by the overlay content paddings
-        const content = this.$.content;
-        const style = getComputedStyle(content);
-
-        // Horizontal adjustment
-        const isLeftAligned = !!this.style.left;
-        if (isLeftAligned) {
-          this.style.left = `${parseFloat(this.style.left) + parseFloat(style.paddingLeft)}px`;
-        } else {
-          this.style.right = `${parseFloat(this.style.right) + parseFloat(style.paddingRight)}px`;
+        this._updatePositionByParentOverlay();
+      } else if (this.positionTarget && this.position) {
+        // This is a root overlay with custom positioning
+        // Adjust position for centering the overlay
+        this._updatePositionByPositionProperty();
+      }
+    }
+
+    /**
+     * @protected
+     * @override
+     */
+    _updatePositionByParentOverlay() {
+      const content = this.$.content;
+      const style = getComputedStyle(content);
+
+      // Horizontal adjustment
+      const isLeftAligned = !!this.style.left;
+      if (isLeftAligned) {
+        this.style.left = `${parseFloat(this.style.left) + parseFloat(style.paddingLeft)}px`;
+      } else {
+        this.style.right = `${parseFloat(this.style.right) + parseFloat(style.paddingRight)}px`;
+      }
+
+      // Vertical adjustment
+      const isBottomAligned = !!this.style.bottom;
+      if (isBottomAligned) {
+        this.style.bottom = `${parseFloat(this.style.bottom) - parseFloat(style.paddingBottom)}px`;
+      } else {
+        this.style.top = `${parseFloat(this.style.top) - parseFloat(style.paddingTop)}px`;
+      }
+    }
+
+    /**
+     * @protected
+     * @override
+     */
+    _updatePositionByPositionProperty() {
+      this._handleOverlayHorizontalCentering();
+      this._handleOverlayVerticalCentering();
+    }
+
+    /**
+     * @protected
+     * @override
+     */
+    _handleOverlayHorizontalCentering() {
```

Comment:
The "centering" term is somewhat confusing as this doesn't necessarily center. It's also not obvious that this method is used only be the position property logic. I'd suggest to inline both methods into `_updatePositionByPositionProperty` since they are not that big.

Reply:
I misread the code, it's actually only for centered positions. Still, I'd prefer inlining the code.